budgetportal/tests/test_provincial_infrastructure_projects.py

```diff
@@ -181,3 +187,162 @@ def test_other_parties(self):
+        department = "Department 1"
+        project = ProvInfraProject.objects.get(department=department)
+
+        data = {"department": department}
+        response = self.client.get(self.url, data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        number_of_projects = len(response.data["results"])
+        response_data = response.data["results"][0]
+        self.assertEqual(number_of_projects, 1)
+        self.assertEqual(response_data["name"], project.name)
+
+    def test_filter_by_province(self):
+        province = "Eastern Cape"
+        data = {"province": province}
+        response = self.client.get(self.url, data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        number_of_projects = len(response.data["results"])
+        self.assertEqual(number_of_projects, 15)
+
+    def test_filter_by_status(self):
+        status_ = "Construction"
+        data = {"status": status_}
+        response = self.client.get(self.url, data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        number_of_projects = len(response.data["results"])
+        self.assertEqual(number_of_projects, 15)
+
+    def test_filter_by_funding_source(self):
+        source = "Community Library Service Grant"
+        data = {"primary_funding_source": source}
+        response = self.client.get(self.url, data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        number_of_projects = len(response.data["results"])
+        self.assertEqual(number_of_projects, 15)
+
+    def test_search_by_project_name(self):
+        name = "Project 1"
+        data = {"search": name}
+        response = self.client.get(self.url, data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertContains(response, name)
+
+    def test_search_by_municipality(self):
+        municipality = "Local 1"
+        data = {"search": municipality}
+        response = self.client.get(self.url, data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertContains(response, municipality)
+
+    def test_search_by_province(self):
+        province = "Eastern Cape"
+        data = {"search": province}
+        response = self.client.get(self.url, data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertContains(response, province)
+
+    def test_search_by_contractor(self):
+        contractor = "Contractor 3"
+        data = {"search": contractor}
+        response = self.client.get(self.url, data)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertContains(response, contractor)
+
+    def test_search_multiple_fields(self):
+        ProvInfraProject.objects.create(
+            financial_year=self.fin_year,
+            IRM_project_id=12345,
+            name="Eastern Cape School",
+        )
+        data = {"search": "Eastern Cape"}
```

Comment:
You should be searching for `Eastern Cape School`. The project shouldn't be named eastern cape school :)

This is verifying the use case of someone too azy to pick a filter, and they just write eastern cape school, and we get a match with a project that has eastern cape in the province field, and school in the name. but the string in the search field is `eastern cape school`

```suggestion
        data = {"search": "Eastern Cape School"}
```